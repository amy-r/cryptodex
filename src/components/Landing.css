@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,700');

.landing {
  position: relative;
  z-index: 10;
  text-align: center;
}

h1 {
  font-family: 'Open Sans', sans-serif;
  letter-spacing: 10px;
  color: white;
}

p {
  font-family: 'Open Sans', sans-serif;
  color: white;
}

.crypto {
  font-weight: 300;
}

.dex {
  font-weight: 700;
}