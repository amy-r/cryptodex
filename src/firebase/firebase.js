import * as firebase from 'firebase';
 
export const config = {
  apiKey: "AIzaSyDxCiaMP6Dc5ApZ1LnmH18xwAr0XnOIBSU",
  authDomain: "cryptoex-be585.firebaseapp.com",
  databaseURL: "https://cryptoex-be585.firebaseio.com",
  projectId: "cryptoex-be585",
  storageBucket: "cryptoex-be585.appspot.com",
  messagingSenderId: "306416025442"
};



export const uiConfig = {
  // Popup signin flow rather than redirect flow.
  signInFlow: 'popup',
  // Redirect to /signedIn after sign in is successful. Alternatively you can provide a callbacks.signInSuccess function.
  signInSuccessUrl: '/home',
  // We will display Google and Facebook as auth providers.
  signInOptions: [
    firebase.auth.GoogleAuthProvider.PROVIDER_ID,
  ]
};

