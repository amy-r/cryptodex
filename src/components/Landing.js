import React from 'react';
import './Landing.css';
import StyledFirebaseAuth from 'react-firebaseui/StyledFirebaseAuth';
import { uiConfig, config } from '../firebase/firebase';
import * as firebase from 'firebase';

const LandingPage = (props) => {

  return (
    <div className = 'landing'>
      <h1> <span className='crypto'>CRYPTO</span><span className='dex'>DEX</span></h1>
      <p> An open-source indexing app for the cryptocurrency market </p>
      <StyledFirebaseAuth uiConfig={uiConfig} firebaseAuth={firebase.auth()} /> 
    </div>
  )
}

export default LandingPage;