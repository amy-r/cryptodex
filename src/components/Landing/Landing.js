import React, { Component } from 'react';
import './Landing.css';
import StyledFirebaseAuth from 'react-firebaseui/StyledFirebaseAuth';
import { uiConfig } from '../../firebase/firebase';
import * as firebase from 'firebase';
import TopTen from '../../containers/TopTen/TopTen';

class Landing extends Component {

  render() {
    return (
      <div className = 'landing'>
        <h1> 
          <span className='crypto'>CRYPTO</span>
          <span className='dex'>DEX</span>
        </h1>
        <p> 
          An open-source indexing app for the cryptocurrency market
        </p>
        <StyledFirebaseAuth 
          uiConfig={uiConfig} 
          firebaseAuth={firebase.auth()} 
        /> 
        <h2> Current Market Index</h2>
        <TopTen />
      </div>
    );
  }
}

export default Landing;