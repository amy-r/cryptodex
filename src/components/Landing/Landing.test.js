/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import Landing from './Landing';
import * as firebase from 'firebase';

describe('Landing', () => {
  it.skip('should match snapshot', () => {
    const wrapper = shallow(<Landing />, {disableLifecycleMethods:true})
    expect(wrapper).toMatchSnapshot();
  })
})