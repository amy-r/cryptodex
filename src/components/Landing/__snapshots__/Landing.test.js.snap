// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          <span
                    className="crypto"
          >
                    CRYPTO
          </span>
          <span
                    className="dex"
          >
                    DEX
          </span>
</h1>,
        <p>
          An open-source indexing app for the cryptocurrency market
</p>,
        <e
          firebaseAuth={undefined}
          uiConfig={
                    Object {
                              "signInFlow": "popup",
                              "signInOptions": Array [
                                "google.com",
                              ],
                              "signInSuccessUrl": "/home",
                            }
          }
/>,
        <h2>
           Today's Market 
</h2>,
        <Connect(Component) />,
      ],
      "className": "landing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="crypto"
>
              CRYPTO
</span>,
            <span
              className="dex"
>
              DEX
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "CRYPTO",
              "className": "crypto",
            },
            "ref": null,
            "rendered": "CRYPTO",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "DEX",
              "className": "dex",
            },
            "ref": null,
            "rendered": "DEX",
            "type": "span",
          },
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "An open-source indexing app for the cryptocurrency market",
        },
        "ref": null,
        "rendered": "An open-source indexing app for the cryptocurrency market",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "firebaseAuth": undefined,
          "uiConfig": Object {
            "signInFlow": "popup",
            "signInOptions": Array [
              "google.com",
            ],
            "signInSuccessUrl": "/home",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Today's Market ",
        },
        "ref": null,
        "rendered": " Today's Market ",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            <span
                        className="crypto"
            >
                        CRYPTO
            </span>
            <span
                        className="dex"
            >
                        DEX
            </span>
</h1>,
          <p>
            An open-source indexing app for the cryptocurrency market
</p>,
          <e
            firebaseAuth={undefined}
            uiConfig={
                        Object {
                                    "signInFlow": "popup",
                                    "signInOptions": Array [
                                      "google.com",
                                    ],
                                    "signInSuccessUrl": "/home",
                                  }
            }
/>,
          <h2>
             Today's Market 
</h2>,
          <Connect(Component) />,
        ],
        "className": "landing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="crypto"
>
                CRYPTO
</span>,
              <span
                className="dex"
>
                DEX
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "CRYPTO",
                "className": "crypto",
              },
              "ref": null,
              "rendered": "CRYPTO",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "DEX",
                "className": "dex",
              },
              "ref": null,
              "rendered": "DEX",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "An open-source indexing app for the cryptocurrency market",
          },
          "ref": null,
          "rendered": "An open-source indexing app for the cryptocurrency market",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "firebaseAuth": undefined,
            "uiConfig": Object {
              "signInFlow": "popup",
              "signInOptions": Array [
                "google.com",
              ],
              "signInSuccessUrl": "/home",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Today's Market ",
          },
          "ref": null,
          "rendered": " Today's Market ",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
